### PR TITLE
[WIP] PR to make IPA work for malicious setting

### PR DESCRIPTION
**Pull request by richajaindce**
I am going to work on a dummy function named "ipa_malicious_wip" to make IPA malicious.

### CODE REVIEW

File: src/protocol/ipa/mod.rs [hunk 1]
```diff
@@ -233,6 +233,111 @@ where
...
+/// # Panics
+/// Propagates errors from multiplications
+#[allow(dead_code)]
+pub async fn ipa_malicious_wip<F, B>(
+    sh_ctx: SemiHonestContext<'_, F>,
+    input_rows: &[IPAInputRow<F, B>],
+    per_user_credit_cap: u32,
+    max_breakdown_key: u128,
+    num_multi_bits: u32,
+) -> Result<Vec<AggregateCreditOutputRow<F>>, Error>
```
Comment (benjaminsavage) [inline src/protocol/ipa/mod.rs:247]: This is now 
```
) -> Result<Vec<MCAggregateCreditOutputRow<F>>, Error>
```

File: src/protocol/ipa/mod.rs [hunk 2]
```diff
@@ -233,6 +233,111 @@ where
...
 }
 
+/// # Errors
+/// Propagates errors from multiplications
+/// # Panics
+/// Propagates errors from multiplications
+#[allow(dead_code)]
+pub async fn ipa_malicious_wip<F, B>(
+    sh_ctx: SemiHonestContext<'_, F>,
+    input_rows: &[IPAInputRow<F, B>],
```
Comment (benjaminsavage) [inline src/protocol/ipa/mod.rs:243]: I think this is now a bit out of date. When I git pull, I see it as:
```
input_rows: &[IPAInputRow<F, MK, BK>],
```

Comment (taikiy) [reply src/protocol/ipa/mod.rs:243]: MK = BitArray trait for match keys
BK = BitArray trait for breakdown keys

@richajaindce ping me if you need help with the new code.

**Comment by richajaindce** [comment]
Closing this and copying code from latest version

**Pull request closed without merging**